src/test/java/org/broadinstitute/hellbender/tools/spark/sv/RunBWAMEMViaCommandLineTest.java

```diff
@@ -0,0 +1,102 @@
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Tests RunBWAMEMViaCommandLine.
+ */
+public final class RunBWAMEMViaCommandLineTest extends CommandLineProgramTest {
+
+    private static final File TEST_DATA_DIR = new File(getTestDataDir(), "spark/sv/RunBWAMEMViaCommandLine");
+    private static final File expectedSam = new File(TEST_DATA_DIR, "expected.sam");
+    private static final Path ref = Paths.get(b37_reference_20_21);
+    private static final Path bwaPath;
+    static {
+        String s = "";
+        try{
+            s = getNonNullEnvironmentVariable("BWA_DIR") + "/bwa";
+        }catch(final UserException e){
+            s = "/usr/local/bin/bwa";
+        }
+        bwaPath = Paths.get(s);
+    }
+
+    @Test(groups="sv")
+    public void testSeparate() throws IOException {
+
+        final ArgumentsBuilder args = new ArgumentsBuilder();
+        final File samOutput = boilerPlate(args);
+
+        // input arguments
+        final File input = new File(TEST_DATA_DIR, "input_1.fastq");
+        final File secondInput = new File(TEST_DATA_DIR, "input_2.fastq");
+        args.add(input.getAbsolutePath());
+        args.add(secondInput.getAbsolutePath());
+
+        this.runCommandLine(args.getArgsArray());
+
+        Assert.assertEquals(getSamRecords(expectedSam), getSamRecords(samOutput));
+    }
+
+    @Test(groups="sv")
+    public void testInterLeaved() throws IOException {
+
+        final ArgumentsBuilder args = new ArgumentsBuilder();
+        final File samOutput = boilerPlate(args);
+
+        // input arguments
+        args.add("-p");
+        final File input = new File(TEST_DATA_DIR, "interleaved.fastq");
+        args.add(input.getAbsolutePath());
+
+        this.runCommandLine(args.getArgsArray());
+
+        Assert.assertEquals(getSamRecords(expectedSam), getSamRecords(samOutput));
+    }
+
+    private static File boilerPlate(final ArgumentsBuilder args) throws IOException{
+
+        args.add("-" + "bwaPath");
+        args.add(bwaPath.toString());
+
+        final File output = Files.createTempFile("test", "sam").toFile();
+        output.deleteOnExit();
+        args.add("-" + StandardArgumentDefinitions.OUTPUT_SHORT_NAME);
+        args.add(output.getAbsolutePath());
+
+        final File REF = ref.toFile();
+        args.add("-" + StandardArgumentDefinitions.REFERENCE_SHORT_NAME);
+        args.add(REF.getAbsolutePath());
+
+        return output;
+    }
+
+    // utility function to extract just alignment records
+    // header information, in particular the @PG line, depends on environment in which the test is run and
+    // in which expected sam is generated, so is un-reproducible across different testing hosts
+    private static List<String> getSamRecords (final File samFile) throws IOException{
+        final List<String> samRecords = new ArrayList<>();
+        try(final BufferedReader reader = new BufferedReader(new FileReader(samFile))){
```

Comment:
[FindBugs] WARNING: Dm: Found reliance on default encoding in org.broadinstitute.hellbender.tools.spark.sv.RunBWAMEMViaCommandLineTest.getSamRecords(File): new java.io.FileReader(File)